src/charm.py

```diff
@@ -147,6 +152,9 @@ def __init__(self, *args):
             substrate="vm",
         )
 
+        self.mysql_logs = MySQLLogs(self)
+        self.framework.observe(self.on.flush_mysql_logs, self.mysql_logs._flush_mysql_logs)
```

Comment:
I think you can move this in the MySQLLogs __init__. You have a reference to the charm and it only calls that class.

Reply:
addressed in 7b72751dc5e7446f48cd1cc8fd5e0ac8e1b7d3db